src/vm/methodtablebuilder.cpp

```diff
@@ -6962,13 +6962,27 @@ MethodTableBuilder::NeedsNativeCodeSlot(bmtMDMethod * pMDMethod)
 
 #ifdef FEATURE_TIERED_COMPILATION
     // Keep in-sync with MethodDesc::DetermineAndSetIsEligibleForTieredCompilation()
-    if (g_pConfig->TieredCompilation() &&
+    if ((g_pConfig->TieredCompilation() &&
 
         // Policy - If QuickJit is disabled and the module does not have any pregenerated code, the method would be ineligible
         // for tiering currently to avoid some unnecessary overhead
         (g_pConfig->TieredCompilation_QuickJit() || GetModule()->HasNativeOrReadyToRunImage()) &&
 
         (pMDMethod->GetMethodType() == METHOD_TYPE_NORMAL || pMDMethod->GetMethodType() == METHOD_TYPE_INSTANTIATED))
+
+#ifdef FEATURE_REJIT
+        ||
```

Comment:
I am not a fan of all these compound && and || clauses.

My preference is to write code that checks one or two things then returns true or false.
And repeat.

So:
```
if (!g_pConfig->TieredCompilation())
{
    return false;
}
 // Policy - If QuickJit is disabled and the module does not have any pregenerated code, the method would       
// be ineligible for tiering currently to avoid some unnecessary overhead
if (!g_pConfig->TieredCompilation_QuickJit() && !GetModule()->HasNativeOrReadyToRunImage())
{
    rerturn false;
}
```